fix for correct sample ids (0-255)!

Frederik-D-Weber opened the pull request:
Scope of the local variable "b"  used in def read(n) was wrong and resulted (at least) in wrong sample ids!
Renameing the variable to "bb" resolved the ambiguity and fixed the problem.

yufengwng commented:
@Frederik-D-Weber can you explain why this will result in wrong sample ids? As far as I can tell, by creating an inner function `read()`, doesn't that create a new scope for the `b` variable? Also, `b` is only referenced when parsing the start byte and not used for parsing the `packet_id` or the other data, so I don't see how it will affect the sample id.

Nonetheless, I agree that having a different variable name will be better for readability.

jfrey-xx commented:
Hi,

I concur with @yufengwng, this patch improves slightly readability even if I don't see the side effect.

Frederik-D-Weber commented:
@yufengwng @jfrey-xx , sorry for your inconvenience, indeed the inner scope was just shadowed by the variable but did not introduce any problems. 

Coincidentally, as I renamed the variable (as suggested by IDE) this seemingly fixed the wrong sample id issue I had while debugging. I switched the variable names between "bb" and "b" at least twice to re-created or abolished the sample id issue back then, thus I thought it might relate to very weird scope bug (which I could not explain myself). I posted a fix in the hope to be enlightened about this weird behavior. However I can confirm now it was just coincidence now, since I failed to recreate the error. Coincidental bug-like behavior is just the worse (-;
